apps/prairielearn/src/sprocs/check_assessment_access.sql

```diff
@@ -26,28 +26,71 @@ DECLARE
     active_access_rule_id bigint;
     next_active_start_date TIMESTAMP WITH TIME ZONE;
     next_active_credit integer;
+    credit_from_override integer;
+    end_date_from_override TIMESTAMP WITH TIME ZONE;
+
 BEGIN
+    -- Check if the user has an entry in the assessment_access_policies table for this assessment_id.
+    -- If yes, get the end_date from the assessment_access_policies table, otherwise, use the end_date from the assessment_access_rules table.
+    SELECT end_date , credit
+    INTO end_date_from_override , credit_from_override
+    FROM assessment_access_policies
+    WHERE assessment_id = check_assessment_access.assessment_id
+        AND (student_uid = check_assessment_access.uid)
+        AND start_date <= check_assessment_access.date
+        AND end_date >= check_assessment_access.date
+    ORDER BY end_date DESC
+    LIMIT 1;
```

Comment:
This doesn't currently take into account group work. We'd want to also check if a student belongs to a group and if that group as a whole has an override. I'll talk with Matt about the best way to handle this.

Reply:
Yes, for now I have only encorporated individual student assessments, as I want uncertain how groups would work